acceptance/tests/connect/local_rate_limit_test.go

```diff
@@ -89,22 +101,21 @@ func assertRateLimits(t *testing.T, opts *assertRateLimitOptions, curlArgs ...st
 	retry.RunWith(opts.retryTimer, t, func(r *retry.R) {
 		// Make up to the allowed numbers of calls in a second
 		t0 := time.Now()
-		for i := 0; i < opts.rps; i++ {
-			output, err := k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, args...)
-			require.NoError(r, err)
-			require.Contains(r, output, opts.successOutput)
-		}
+
+		output, err := k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, append(args, repeatAddr)...)
+		require.NoError(r, err)
+		require.Contains(r, output, opts.successOutput)
 
 		// Exceed the configured rate limit.
-		output, err := k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, args...)
+		output, err = k8s.RunKubectlAndGetOutputE(t, opts.k8sOpts, append(args, addr)...)
+		require.True(r, time.Since(t0) < time.Second, "failed to make all requests within one second window")
 		if opts.enforced {
 			require.Error(r, err)
 			require.Contains(r, output, opts.rateLimitOutput, "request was not rate limited")
 		} else {
 			require.NoError(r, err)
-			require.Contains(r, output, opts.successOutput, "request was not successful")
+			require.NotContains(r, output, opts.rateLimitOutput, "request was not successful")
```

Comment:
Since we are globbing requests we get all the outputs of successes and failures combined; we are better off making sure that 429s are NOT found when rate limiting is not enforced